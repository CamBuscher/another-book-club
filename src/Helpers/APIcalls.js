import googleKey from './APIkey';

export const searchViaAuthorTitleISBN = async (query) => {
  const response = await fetch(`https://www.googleapis.com/books/v1/volumes?key=${googleKey}&q=${query}`);
  console.log(response.json());
};

export const searchFreeEbooksByGenre = async (genre) => {
  const response = await fetch(`https://www.googleapis.com/books/v1/volumes?key=AIzaSyDwf3i034XmIqxzvCyEhafOTlMlTscHNNE&filter=free-ebooks&q=${genre}`);
  const results = await response.json();
  const booksInfo = results.items.map(item => item.volumeInfo);
  return booksInfo;
};

