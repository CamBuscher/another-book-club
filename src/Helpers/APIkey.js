const goodreadsKEY = 'qTOyahcP70E436vVqi3fQ'

export default goodreadsKEY