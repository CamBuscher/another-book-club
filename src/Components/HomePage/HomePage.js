import React, { Component } from 'react';
import * as APIcalls from '../../Helpers/APIcalls';
import { BooksDisplay } from '../BooksDisplay/BooksDisplay';
import PropTypes from 'prop-types';

class HomePage extends Component {
  constructor(props) {
    super(props);

    this.state = {
      searchValue : '',
      searchResults : []
    };
  }

  handleInputSearch = async (e) => {
    e.preventDefault();
    const searchResults = await APIcalls.searchViaAuthorTitleISBN(this.state.searchValue);
    this.setState({ searchResults });
  }

  render() {
    const isInvalid = this.state.searchValue === '';

    return (
      <div>
        <h5>Search for a book</h5>
        <form onSubmit={this.handleInputSearch}>
          <input 
            placeholder='Search for some books' 
            value={this.state.searchValue} 
            onChange={(e) => { this.setState({searchValue : e.target.value}); }}
          />
          <button type='submit' disabled={isInvalid} >Search</button>
        </form>
        <BooksDisplay 
          books={this.state.searchResults} 
          currentClub={this.props.currentClub}
          addBookToClub={this.props.addBookToClub}
        />
      </div>
    );
  }
}

HomePage.propTypes = {
  currentClub: PropTypes.object,
  addBookToClub: PropTypes.func
};

export default HomePage;