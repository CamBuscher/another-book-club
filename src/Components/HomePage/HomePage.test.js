import React from 'react';
import { shallow } from 'enzyme';
import HomePage from './HomePage';
import * as APIcalls from '../../Helpers/APIcalls';

describe('HomePage', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<HomePage />);
    APIcalls.searchFreeEbooksByGenre = jest.fn().mockImplementation(() => ['dog']);
    APIcalls.searchViaAuthorTitleISBN = jest.fn().mockImplementation(() => ['dog']);
  });

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should update state when a user types in input field', () => {
    const input = wrapper.find('input');
    
    input.simulate('change', { target: {
      value: 'hello'
    }});

    expect(wrapper.state().searchValue).toEqual('hello');
  });

  describe('handleInputSearch', () => {
    it('should call the search via Author/title API call', async () => {
      const form = wrapper.find('form');
      const input = wrapper.find('input');

      input.simulate('change', {
        target: {
          value: 'hello'
        }
      });

      await form.simulate('submit', {preventDefault: jest.fn()});

      expect(APIcalls.searchViaAuthorTitleISBN).toHaveBeenCalledWith('hello');
    });

    it('should set the state with search results', async () => {
      const expected = ['dog'];
      APIcalls.searchViaAuthorTitleISBN = jest.fn().mockImplementation(() => ['dog']);
      const form = wrapper.find('form');
      const input = wrapper.find('input');

      input.simulate('change', {
        target: {
          value: 'hello'
        }
      });

      await form.simulate('submit', { preventDefault: jest.fn() });

      expect(wrapper.state().searchResults).toEqual(expected);
    });
  });
});