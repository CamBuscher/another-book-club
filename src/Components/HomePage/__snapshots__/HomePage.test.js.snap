// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomePage should match snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <HomePage />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h5>
          Search for a book
</h5>,
        <form
          onSubmit={[Function]}
>
          <input
                    onChange={[Function]}
                    placeholder="Search for some books"
                    value=""
          />
          <button
                    disabled={true}
                    type="submit"
          >
                    Search
          </button>
</form>,
        <Unknown
          addBookToClub={undefined}
          books={Array []}
          currentClub={undefined}
/>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Search for a book",
        },
        "ref": null,
        "rendered": "Search for a book",
        "type": "h5",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <input
              onChange={[Function]}
              placeholder="Search for some books"
              value=""
/>,
            <button
              disabled={true}
              type="submit"
>
              Search
</button>,
          ],
          "onSubmit": [Function],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "onChange": [Function],
              "placeholder": "Search for some books",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Search",
              "disabled": true,
              "type": "submit",
            },
            "ref": null,
            "rendered": "Search",
            "type": "button",
          },
        ],
        "type": "form",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "addBookToClub": undefined,
          "books": Array [],
          "currentClub": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h5>
            Search for a book
</h5>,
          <form
            onSubmit={[Function]}
>
            <input
                        onChange={[Function]}
                        placeholder="Search for some books"
                        value=""
            />
            <button
                        disabled={true}
                        type="submit"
            >
                        Search
            </button>
</form>,
          <Unknown
            addBookToClub={undefined}
            books={Array []}
            currentClub={undefined}
/>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Search for a book",
          },
          "ref": null,
          "rendered": "Search for a book",
          "type": "h5",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <input
                onChange={[Function]}
                placeholder="Search for some books"
                value=""
/>,
              <button
                disabled={true}
                type="submit"
>
                Search
</button>,
            ],
            "onSubmit": [Function],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "onChange": [Function],
                "placeholder": "Search for some books",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Search",
                "disabled": true,
                "type": "submit",
              },
              "ref": null,
              "rendered": "Search",
              "type": "button",
            },
          ],
          "type": "form",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "addBookToClub": undefined,
            "books": Array [],
            "currentClub": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
